skorch/tests/test_utils.py

```diff
@@ -485,3 +486,28 @@ def type_truth_table():
         type_truth_table())
     def test_data_types(self, is_skorch_dataset, input_data, expected):
         assert is_skorch_dataset(input_data) == expected
+
+
+class TestTeeGenerator:
+
+    @pytest.fixture
+    def lazy_generator_cls(self):
+        from skorch.utils import TeeGenerator
+        return TeeGenerator
+
+    def test_calls_generator_once(self, lazy_generator_cls):
+        expected_list = [1, 2, 3]
+
+        def list_gen():
+            yield from expected_list
+        mock = Mock(return_value=list_gen())
+        lazy_gen = lazy_generator_cls(mock)
+
+        assert mock.call_count == 0
+
+        first_return = list(lazy_gen)
```

Comment:
The test would be a little clearer if the call count would also be tested immediately after the first call (i.e. here).